decaf377-ka/src/lib.rs

```diff
@@ -0,0 +1,52 @@
+use ark_ff::UniformRand;
+
+use decaf377;
+use rand_core::{CryptoRng, RngCore};
+
+/// A `SharedSecret` derived at the end of the key agreement protocol.
+#[derive(Debug, PartialEq)]
+pub struct SharedSecret(pub(crate) decaf377::Element);
+
+impl SharedSecret {
+    pub fn derive(other_public_key: &EphemeralPublicKey, secret_key: EphemeralSecretKey) -> Self {
```

Comment:
I think the API would be more ergonomic if this were

```rust
impl EphemeralSecretKey {
    pub fn key_agreement_with(self, other: &EphemeralPublicKey) -> SharedSecret { ... }
}
```

or some other name, we could bikeshed it a little bit, but the important thing is that by making the consumed secret be a `self` parameter, there's a clearer distinction between the two parameters: I use my secret key with the counterparty's public key as `my_secret.key_agreement_with(other_pub)` or something, rather than having to remember which parameter comes first (is it public/secret or secret/public?). 

It's also more common to have APIs that consume `self` as a way to enforce state transitions, so I think that's also a bit more familiar.